services/gateway/all-area-broadcaster.mjs

```diff
@@ -81,30 +80,30 @@ class AllAreaBroadcaster {
   }
 
   /**
-   * This is called when we've a location message from redis.
-   * @param {object} loc - The location object, see Gateway Service for doc.
+   * Register onLocation function.
+   * @param {Function} callback - This is called when we've a location
+   * message from redis.
    */
-  onLocation(loc) {
-    // Broadcast the location message.
-    this.io.emit('location', loc);
+  registerOnLocation(callback) {
+    this.onLocation = callback;
```

Comment:
You might need to do `this.onLocation = () => {};` in the constructor to make sure even if registerOnLocation() is not called, we don't crash.

Same goes to others below.